composer/callbacks/speed_monitor.py

```diff
@@ -124,3 +181,15 @@ def batch_end(self, state: State, logger: Logger):
     def eval_end(self, state: State, logger: Logger):
         del logger  # unused
         self.total_eval_wct += state.eval_timestamp.total_wct.total_seconds()
+        assert state.dataloader_label is not None, 'evaluator label must not be None'
```

Comment:
QQ: When can eval dataloader label be none? if it can only be none when you only have one evaluator, can you just use a default name in the none case?

Reply:
It can be `None` if user explicitly sets it to None when constructing it. This violates type hints and breaks things iirc, it really should always be set. However, `state.dataloader_label` can be `None` as it is possible to have no active dataloader, so we need this check for lint reasons (logically it should never happen)

Reply:
got it, can you change the comment on the assert to say its for lint reasons?